homeassistant/components/geniushub/climate.py

```diff
@@ -0,0 +1,159 @@
+        if not hasattr(self._objref, 'occupied'):
+            op_list.remove(STATE_ECO)
+        self._operation_list = op_list
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._objref.name
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes of the evohome Climate device.
+
+        This is state data that is not available otherwise, due to the
+        restrictions placed upon ClimateDevice properties, etc. by HA.
+        """
+        tmp = self._objref.__dict__.items()
+        state = {k: v for k, v in tmp if k in GH_DEVICE_STATE_ATTRS}
+
+        return {'status': state}
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._objref.temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._objref.setpoint
+
+    @property
+    def min_temp(self):
+        """Return max valid temperature that can be set."""
+        return GENIUSHUB_MIN_TEMP
+
+    @property
+    def max_temp(self):
+        """Return max valid temperature that can be set."""
+        return GENIUSHUB_MAX_TEMP
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return GENIUSHUB_SUPPORT_FLAGS
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        return GH_STATE_TO_HA.get(self._objref.mode, None)
+
+    @property
+    def is_on(self):
+        """Return True if the device is on."""
+        return self._objref.mode in GH_STATE_TO_HA
+
+    async def async_set_operation_mode(self, operation_mode):
+        """Set a new operation mode for this zone."""
+        await self._objref.set_mode(HA_OPMODE_TO_GH.get(operation_mode))
+
+    async def async_set_temperature(self, **kwargs):
+        """Set a new target temperature for this zone."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        await self._objref.set_override(temperature, 3600)  # 1 hour
+
+    async def async_turn_on(self):
+        """Turn on this heating zone."""
+        await self._objref.set_mode(HA_OPMODE_TO_GH.get(STATE_AUTO))
+
+    async def async_turn_off(self):
+        """Turn off this heating zone (i.e. to frost protect)."""
+        await self._objref.set_mode(GH_OPMODE_OFF)
+
+    async def async_update(self):
+        """Get the latest data from the hub."""
+        try:
+            await self._objref.update()
+        except (AssertionError, asyncio.TimeoutError) as err:
+            _LOGGER.warning("self.(%s).update(): Failed "
```

Comment:
Just write that "Update for [name] failed", and include the error message. Don't include a vague assumption/question.

Reply:
Resolved.